frontend/src/pages/users/+Page.tsx

```diff
@@ -0,0 +1,232 @@
+import { useAuthContext } from '../../components/useAuthContext'
+import { useState, useEffect } from 'react'
+import styled from 'styled-components'
+
+interface User {
+  email: string
+  is_admin: boolean
+  is_manager: boolean
+  id: number
+}
+
+const Container = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  margin-top: 50px;
+`
+
+const Table = styled.table`
+  width: 80%;
+  margin-bottom: 20px;
+`
+
+const Th = styled.th`
+  padding: 10px;
+  text-align: left;
+  border-bottom: 1px solid #ccc;
+`
+
+const Td = styled.td`
+  padding: 10px;
+  border-bottom: 1px solid #ccc;
+`
+
+const Input = styled.input`
+  margin-bottom: 10px;
+  padding: 5px;
+`
+
+const Button = styled.button`
+  padding: 8px 16px;
+  margin-top: 10px;
+`
+
+const ErrorTag = styled.p`
+  color: red
+`
+
+function Page(): JSX.Element {
+  const authContext = useAuthContext()
+
+  const [users, setUsers] = useState<User[]>([])
+  const [newUser, setNewUser] = useState<{ email: string, isAdmin: boolean, isManager: boolean }>({ email: '', isAdmin: false, isManager: false })
+  const [error, setError] = useState<string>('')
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const { name, value } = e.target
+    setNewUser({ ...newUser, [name]: value })
+  }
+
+  useEffect(() => {
+    if (authContext.isAdmin) {
+      fetchUsers().catch((error) => {
+        console.error('Fetch error', error)
+      })
+    }
+  }, [authContext.isAdmin])
+
+  const fetchUsers = async (): Promise<void> => {
+    const response = await fetch('/api/user')
+    if (!response.ok) {
+      throw new Error('Failed to get users')
+    }
+    const data: User[] = await response.json()
+    console.log(data)
+    setUsers(data)
+  }
+
+  const addUser = async (): Promise<void> => {
+    const { email, isAdmin, isManager } = newUser
+
+    if (email.match(/^\S+@rice\.edu$/) == null) {
+      setError('Please enter a valid Rice University email')
+      return
+    }
+    try {
+      const response = await fetch('/api/user', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({ email, is_admin: isAdmin, is_manager: isManager })
+      })
+
+      if (!response.ok) {
+        setError('Failed to add user')
+        throw new Error('Failed to add user')
+      }
+
+      const data: User = await response.json()
+      setUsers([...users, data])
+      setNewUser({ email: '', isAdmin: false, isManager: false })
+      setError('')
```

Comment:
:thinking: -- Why is this code not being covered by `adds a user when addUser is called`?